### ISSUE TITLE
Support for MacOS Unified Logging  

### ISSUE THREAD

**Issue by nathanrstacey** [elastic/integrations#6589]
**Describe the enhancement:**
Support for MacOS Unified Logging

**Describe a specific use case for the enhancement or feature:**
Auditbeat doesn’t provide much valuable information because it is still pulling information from syslog, MacOS is deprecating use of syslog and has moved to Unified Logging.  To my knowledge there is no Beat for MacOS that will track login, logout, lock, unlock, or sudo access.  It is possible to create custom scripts to grab some (not all) of this info but a Beat would be much easier

**Comment by defensivedepth** [comment]
I'll just mention that there is a rust library put out by Mandiant to parse the Unified Log - https://github.com/mandiant/macos-UnifiedLogs

**Comment by DefSecSentinel** [comment]
What is the real need here for the Apple Unified Logs? What would the goal be with these logs as opposed to what our macOS agent already provides? Our macOS agent taps into the Apple Endpoint Security Framework (ESF:https://developer.apple.com/documentation/endpointsecurity) which allows us to collect just about anything you could want to access in the Unified Logs via dedicated events vs a filtered event stream. 

From a security perspective collecting Apple Unified Logging provides no real benefit aside from maybe in an incident response capacity collecting data after the fact. In addition to the many ESF events and custom/proprietary data sources our agent provides you can also implement our OSQuery integration with the agent to conduct and collect live queries of absolutely any data source macOS makes available via Unified Logs. 

@defensivedepth Many of the use cases you pose from the Mandiant and Crowdstrike blogs are already covered by what our macOS agent provides utilizing Apple's ESF, or can be queried using OSQuery. The only events our agent doesn't currently collect are login/logout, session lock/unlock, screenattach, and session create/destroy all of which though we could collect and add to our agent with ESF if needed. These events though are more admin/policy focused and less helpful in actively detecting threat actor activity in real time. The events would be useful for Incident Response though or organization specific policy monitoring but like I said you can query this specific data via our agent OSQuery integration currently and we are adding the macOS login/authentication events (found here: https://developer.apple.com/documentation/endpointsecurity/endpointsecurity_structures) to our agent very soon (Apple Ventura Release).

**Comment by defensivedepth** [comment]
@DefSecSentinel 

`Our macOS agent taps into the Apple Endpoint Security Framework (ESF:https://developer.apple.com/documentation/endpointsecurity) which allows us to collect just about anything you could want to access in the Unified Logs via dedicated events vs a filtered event stream.`

What specific Integration are you referring to?

**Comment by DefSecSentinel** [comment]
macOS agent with the Elastic Defend integration installed

**Comment by defensivedepth** [comment]
Can you post a link to the Elastic docs for Defend and ESF? 

I am running Elastic Agent + Defend on macOS and appear to just have File + Process + Network events:

![image](https://github.com/elastic/integrations/assets/954732/5d6350cd-3664-4584-996e-863f728f467e)

**Comment by DefSecSentinel** [comment]
As long as you have those 3 options checked you should be getting everything. What specifically are you looking for?

**Comment by defensivedepth** [comment]
Ok, so to make sure I understand - if I want to detect when a new local user has been created or a configuration profile has been installed, I would need to get that through the File/Process/Network events? The ESF integration you are talking about doesn't create structured logs -  for instance, EventID 4720 in the Security Event Channel on Windows is generated when a new local user is created.

**Comment by DefSecSentinel** [comment]
Yes, there are a number of different queries or detections you could write to see whether a new user has been created or configuration profile has been added using process and file events. 

ESF isn't an Elastic integration. It's a framework provided by Apple on macOS that allows our agent to subscribe to specific events (similar to how you can subscribe to events by event codes like 4720 on Windows). Create user is one of those ESF events (https://developer.apple.com/documentation/endpointsecurity/es_event_od_create_user_t) we could subscribe to but don't currently. 

Are you wanting to know in general if a new user gets created for policy reasons or if a new user gets created in a suspicious or abnormal manner?

**Comment by natewalck** [comment]
For this particular issue, I think ingesting logs from the unified log system is the ask. You can get some info from EndpointSecurity framework, but its not the same as the log data.

**Comment by gord-lyr** [comment]
Any update on this? 

I'm asking because we have a customer making use of this MacOS app: https://github.com/SAP/macOS-enterprise-privileges. Logs for this get pushed to Apple Unified Log, and not sure if this is something we could pickup automatically or not.

There are options for Syslog, but this is only useful for devices that always have line-of-sight to the syslog server, which won't always be the case - so AUL would be the best method.

Thanks

**Comment by leandrojmp** [comment]
Hello, 

We also would like to be able to get **logs** from MacOS as we do not use Elastic Defend and have no plan to use it since we already have other XDR tool.

So since the Elastic Agent cannot get logs form MacOS we have a gap in visbility as we can get logs from Windows and Linux system, but not from macOS.

The user needs to be able to get logs from MacOS in the same way they can from Windows and Linux.

**Comment by nimarezainia** [comment]
fyi @jamiehynds

**Comment by marc-gr** [comment]
Recently a macOS UL input [has been added to filebeat](https://github.com/elastic/beats/pull/41791) and will be available from 8.18.

At the same time, an [input package](https://github.com/elastic/integrations/pull/12096) has been added for it and will be, also, available from 8.18.

We have plans to add a macOS package including some dashboards and pipelines to ingest common use cases similar to the windows package in the near future, I am not sure if we want to keep this open for that purpose or maybe open a more specific issue for it with more concrete requirements.

**Comment by marc-gr** [comment]
> Hello [@marc-gr](https://github.com/marc-gr) , any update on this?
> 
> Does the new macOS Unified Log integration the one that we should use?

That's correct; this is currently the approach for retrieving logs from macOS unified logging. We may implement a dedicated macOS integration for common scenarios in the future, but it's not on our immediate roadmap. I'm closing this issue now and will open a new, specific issue for that integration once we have a clearer direction.
